nuitka/tools/quality/auto_format/YamlFormatter.py

```diff
@@ -1,418 +1,306 @@
+    global OPTIONS_KEYS, OPTIONS_CHECKS_KEYS
+
+    with openTextFile(
+        getYamlPackageConfigurationSchemaFilename(),
+        "r",
+    ) as schema_file:
+        schema = json.load(schema_file)
+
+    MASTER_KEYS = tuple(schema["items"]["properties"].keys())
+    DATA_FILES_KEYS = tuple(
+        schema["items"]["properties"]["data-files"]["properties"].keys()
+    )
+    DLLS_KEYS = tuple(
+        schema["items"]["properties"]["dlls"]["items"]["properties"].keys()
+    )
+    DLLS_BY_CODE_KEYS = tuple(
+        schema["items"]["properties"]["dlls"]["items"]["properties"]["by_code"][
+            "properties"
+        ].keys()
+    )
+    DLLS_FROM_FILENAMES_KEYS = tuple(
+        schema["items"]["properties"]["dlls"]["items"]["properties"]["from_filenames"][
+            "properties"
+        ].keys()
+    )
+    ANTI_BLOAT_KEYS = tuple(
+        schema["items"]["properties"]["anti-bloat"]["items"]["properties"].keys()
+    )
+    IMPLICIT_IMPORTS_KEYS = tuple(
+        schema["items"]["properties"]["implicit-imports"]["items"]["properties"].keys()
+    )
+    OPTIONS_KEYS = tuple(schema["items"]["properties"]["options"]["properties"].keys())
+    OPTIONS_CHECKS_KEYS = tuple(
+        schema["items"]["properties"]["options"]["properties"]["checks"]["items"][
+            "properties"
+        ].keys()
+    )
+
+
+def _decideStrFormat(string):
+    """
+    take the character that is not closest to the beginning or end
+    """
+    # Singleton, pylint: disable=too-many-return-statements
+    if (
+        string not in MASTER_KEYS
+        and string not in DATA_FILES_KEYS
+        and string not in DLLS_KEYS
+        and string not in DLLS_BY_CODE_KEYS
+        and string not in DLLS_FROM_FILENAMES_KEYS
+        and string not in ANTI_BLOAT_KEYS
+        and string not in IMPLICIT_IMPORTS_KEYS
+        and string not in OPTIONS_KEYS
+        and string not in OPTIONS_CHECKS_KEYS
+    ):
+        single_quote_left = string.find("'")
+        single_quote_right = string.rfind("'")
+        quote_left = string.find('"')
+        quote_right = string.rfind('"')
+
+        if single_quote_left == -1 and not quote_left == -1:
+            return "'"
+
+        elif quote_left == -1 and not single_quote_left == -1:
+            return '"'
+
+        elif (
+            single_quote_left == -1
+            and single_quote_right == -1
+            and quote_left == -1
+            and quote_right == -1
+        ):
+            return '"'
+
+        elif single_quote_left > quote_left and single_quote_right < quote_right:
+            return "'"
+
+        elif single_quote_left < quote_left and single_quote_right > quote_right:
+            return '"'
+
+        else:
+            return '"'
+
+    else:
+        return ""
+
+
+class CustomConstructor(ruamel.yaml.constructor.RoundTripConstructor):
+    def construct_scalar(self, node):
+        # Singleton, pylint: disable= too-many-branches
,too-many-return-statements
```

Comment:
Das Rote kann ich lokal in VS Code gar nicht sehen. Keine Ahnung woher das kommt.